plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/util/SymbolIcons.java

```diff
@@ -0,0 +1,108 @@
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.protectedField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.protectedField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.protectedField()))),
+            put(
+                Visibility.PACKAGE,
+                map(
+                    put(SymbolKind.Package, JavaResources.INSTANCE.packageItem()),
+                    put(SymbolKind.Class, JavaResources.INSTANCE.svgClassItem()),
+                    put(SymbolKind.Method, JavaResources.INSTANCE.defaultMethod()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.Constructor, JavaResources.INSTANCE.defaultMethod()),
+                    put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem()),
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.defaultField()))),
+            put(
+                Visibility.PRIVATE,
+                map(
+                    put(SymbolKind.Package, JavaResources.INSTANCE.packageItem()),
+                    put(SymbolKind.Class, JavaResources.INSTANCE.svgClassItem()),
+                    put(SymbolKind.Method, JavaResources.INSTANCE.privateMethod()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.Constructor, JavaResources.INSTANCE.privateMethod()),
+                    put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem()),
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.privateField()))));
+  }
+
+  public SVGResource get(ExtendedSymbolInformation symbol) {
+    Visibility visiblity = symbol.getVisiblity();
+    return resources
+        .get(visiblity == null ? Visibility.PACKAGE : visiblity)
+        .get(symbol.getInfo().getKind());
+  }
+
+  private static <K, V> Consumer<Map<K, V>> put(K key, V value) {
+    return new Consumer<Map<K, V>>() {
+
+      @Override
+      public void accept(Map<K, V> map) {
+        map.put(key, value);
+      }
+    };
+  }
+
+  @SafeVarargs
+  private static <K, V> Map<K, V> map(Consumer<Map<K, V>>... entries) {
+    HashMap<K, V> result = new HashMap<>(entries.length);
+    for (Consumer<Map<K, V>> entry : entries) {
+      entry.accept(result);
+    }
+    return result;
+  }
+}
```

Comment:
In general `put` and `map` methods looks extra and confusing to me. Since we have [google guava](https://github.com/google/guava) as a dependency and it is quite commonly used in many projects including this one, I believe there is no need to reinvent the wheel. I'm wondering why can't we use [`ImmutableMap`](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.html) and [`Builder`](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html) which are designed exactly for such cases and will make the code clearer and save us from unnecessary methods? Please see the example of modified static initialization block:
  ```java
  
  static {
    resources =
        new Builder<Visibility, Map<SymbolKind, SVGResource>>()
            .put(
                Visibility.PUBLIC,
                new Builder<SymbolKind, SVGResource>()
                    .put(SymbolKind.Package, JavaResources.INSTANCE.packageItem())
                    .put(SymbolKind.Class, JavaResources.INSTANCE.svgClassItem())
                    .put(SymbolKind.Method, JavaResources.INSTANCE.publicMethod())
                    .put(SymbolKind.Field, JavaResources.INSTANCE.publicField())
                    .put(SymbolKind.Constructor, JavaResources.INSTANCE.publicMethod())
                    .put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem())
                    .put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem())
                    .put(SymbolKind.Field, JavaResources.INSTANCE.publicField())
                    .put(SymbolKind.Constant, JavaResources.INSTANCE.publicField())
                    .put(SymbolKind.EnumMember, JavaResources.INSTANCE.publicField())
                    .build())
            ...
            .build();
  }
  
  ```